### PR TITLE
str: proper titlecase support

### PR DESCRIPTION
**Pull request by OddCoincidence**
Follow-up to #827 that adds proper titlecase support using a new crate https://github.com/OddCoincidence/unicode-casing.

### CODE REVIEW

File: vm/Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ regex = "1"
 rustc_version_runtime = "0.1.*"
 statrs = "0.10.0"
 caseless = "0.2.1"
+unicode-casing = { git = "https://github.com/OddCoincidence/unicode-casing" }
```
Comment (windelbouwman) [inline vm/Cargo.toml]: Could you create a release from this package, so we can refer to a pinned version?

File: vm/src/obj/objstr.rs [hunk 2]
```diff
@@ -413,12 +414,12 @@ impl PyString {
         for c in self.value.chars() {
             if c.is_lowercase() {
                 if !previous_is_cased {
-                    title.extend(c.to_uppercase());
+                    title.extend(c.to_titlecase());
```
Comment (windelbouwman) [inline vm/src/obj/objstr.rs]: How can a single character be title cases? I guess something else is meant than python's titlecase function?

Comment (OddCoincidence) [reply vm/src/obj/objstr.rs]: This surprised me too, but there indeed exist single codepoint titlecase characters (I realized this from reading the cpython tests for `title` / `istitle`). [Some examples](https://github.com/OddCoincidence/unicode-casing/blob/master/src/lib.rs#L104-L105).

File: vm/Cargo.toml (hunk 1 above)
Comment (OddCoincidence) [reply vm/Cargo.toml]: You mean on crates.io? I was going to see if the rust unicode folks wanted control over this package before submitting. It's fine if you want to wait for that, but I can also add the commit sha here if that makes you feel better.

Comment (windelbouwman) [reply vm/Cargo.toml]: Ah, okay, yes, rust unicode group having this package would be good. Let's keep it like this, until this is effect. Maybe add a todo?

File: vm/src/obj/objstr.rs (hunk 2 above)
Comment (windelbouwman) [reply vm/src/obj/objstr.rs]: "Latin Capital Letter D with Small Letter Z with Caron". Unicode is never boring! :)

**Pull request merged**